test/modules/Modules.Test/TestResultCoordinator/Reports/DirectMethod/MockNetworkStatusTimeline.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Modules.Test.TestResultCoordinator.Reports.DirectMethod
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using global::TestResultCoordinator.Reports;
+    using global::TestResultCoordinator.Reports.DirectMethod;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.NetworkController;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
+    using Microsoft.Azure.Devices.Edge.Storage;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Moq;
+
+    /// <summary>
+    /// This is a helper class for DirectMethodReportGeneratorTest. It simply creates a static mocked NetworkStatusTimeline.
+    /// It's not so easy to mock the timeline, so to avoid clutter in the test class we build the mock here.
+    /// </summary>
+    class MockNetworkStatusTimeline
+    {
+        public static async Task<Option<NetworkStatusTimeline>> GetMock(TimeSpan tolerancePeriod)
+        {
+            return await GetNetworkStatusTimeline(
+                GetNetworkControllerStoreTestResultCollection(
+                    NetworkControllerStatusArray,
+                    DateTimeArray,
+                    NetworkControllerOperationArray),
+                tolerancePeriod);
+        }
+
+        static DateTime[] DateTimeArray => new[]
+        {
+            new DateTime(2020, 1, 1, 9, 10, 10, 10),
+            new DateTime(2020, 1, 1, 9, 10, 10, 13),
+            new DateTime(2020, 1, 1, 9, 10, 15, 10),
+            new DateTime(2020, 1, 1, 9, 10, 15, 13),
+            new DateTime(2020, 1, 1, 9, 10, 20, 10),
+            new DateTime(2020, 1, 1, 9, 10, 20, 13),
+            new DateTime(2020, 1, 1, 9, 10, 25, 10),
+            new DateTime(2020, 1, 1, 9, 10, 25, 13)
+        };
+
+        static string[] NetworkControllerStatusArray => new[] { "Disabled", "Disabled", "Enabled", "Enabled", "Disabled", "Disabled", "Enabled", "Enabled" };
+
+        static string[] NetworkControllerOperationArray => new[] { "SettingRule", "RuleSet", "SettingRule", "RuleSet", "SettingRule", "RuleSet", "SettingRule", "RuleSet" };
+
+        static async Task<Option<NetworkStatusTimeline>> GetNetworkStatusTimeline(StoreTestResultCollection<TestOperationResult> results, TimeSpan tolerancePeriod)
+        {
+            try
+            {
+                return Option.Some(await NetworkStatusTimeline.Create(results, tolerancePeriod));
+            }
+            catch (Exception)
```

Comment:
GetNetworkStatusTimeline is private, you manage all usages.  why do you need to catch exception?

Reply:
Remnant from moving it around. Removed the try catch